src/mac/process.rs

```diff
@@ -8,6 +8,51 @@ use std::fmt::{self, Formatter, Debug};
 use libc::{c_int, gid_t, kill, pid_t, uid_t};
 
 #[derive(Clone)]
+pub enum ProcessStatus {
```

Comment:
See? That's what I said: your code for `ProcessStatus` is exactly the same in both linux and mac so better put it at a higher level to avoid implementing it twice.